File: src/main/java/de/maggiwuerze/xdccloader/model/Channel.java
package de.maggiwuerze.xdccloader.model;

import java.time.LocalDateTime;
import lombok.Data;
import javax.persistence.*;

@Data
@Entity
public class Channel {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        Long id;

        @Column(nullable = false)
        String name;

        @Column(nullable = false)
        LocalDateTime date = LocalDateTime.now();

        public Channel() {
        }

        public Channel(String name) {
                this.name = name;

        }

        public Long getId() {
                return id;
        }

        public void setId(Long id) {
                this.id = id;
        }

        public String getName() {
                return name;
        }

        public void setName(String name) {
                this.name = name;
        }

        public LocalDateTime getDate() {
                return date;
        }

        public void setDate(LocalDateTime date) {
                this.date = date;
        }
}

File: src/main/kotlin/de/maggiwuerze/xdccloader/model/Channel.kt
package de.maggiwuerze.xdccloader.model;

import java.time.LocalDateTime;
import lombok.Data;
import javax.persistence.*

@Entity
data class Channel(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long,

        @Column(nullable = false)
        var name: String,

        @Column(nullable = false)
        var date: LocalDateTime

)